### PR TITLE
Build system and compiler version maintenance

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
Requires https://github.com/sociomantic-tsunami/beaver/pull/26

**Comment by mihails-strasuns-sociomantic** [comment]
Will stay at 2.071 until we figure out what went wrong with allocators in 2.072

File: .travis.yml [hunk 1]
```diff
@@ -19,12 +19,11 @@ env:
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
     matrix:
-        - DIST=trusty
-        - DIST=xenial
+        - DMD='2.070.2-*' DIST=xenial
```
Comment (leandro-lucarella-sociomantic) [inline .travis.yml]: This wildcarding of version is not supported yet.

**Review by leandro-lucarella-sociomantic — COMMENTED** [review]
The version wildcard seems to be working accidentally. I'm not sure if `apt` undestands it directly (I doubt it), or if just an upgrade is triggered because the package is specified and there is a new version in the repo.

In any case, this should be fixed: https://github.com/sociomantic-tsunami/cachalot/issues/23

File: .travis.yml (hunk 1 above)
Comment (mihails-strasuns-sociomantic) [reply .travis.yml]: It is done by `apt` itself. I can't find reference in documentation but I did some manual experiments and am reasonably sure it just instals latest available package where version string matches.

Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: OK, this is very interesting and useful. I was just playing with pinning and repo priorities and it seems to work well with that too. I guess we can move forward with it!

**Review by leandro-lucarella-sociomantic — APPROVED** [review]

**Pull request merged**